app/main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+class Animal:
+    alive: list = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @classmethod
+    def bite(cls, herbivore: Herbivore) -> None:
+        if not isinstance(herbivore, cls) and not herbivore.hidden:
```

Comment:
The condition `not isinstance(herbivore, cls)` is incorrect. It should check if `herbivore` is an instance of `Herbivore`, not `cls`. Change it to `isinstance(herbivore, Herbivore)`.